### PR TITLE
updates to ml diagram and docs

### PR DESCRIPTION
**Pull request by baykaner**

### CODE REVIEW

File: docs/source/ml/index.rst [hunk 1]
```diff
@@ -22,27 +22,42 @@ machine learning frameworks.
...
 
-.. raw:: html
-    :file: ml_lib_overview.svg
 
 
-Important Notes for Working with the Machine Learning Library
+General Notes for Working with the Machine Learning Library
 -------------------------
 
 - Every Op must have a DESCRIPTOR (this is for logging/error reporting)
```
Comment (MissingNO57) [inline docs/source/ml/index.rst]: DESCRIPTOR is also used if you don't specify the name of the node.

**Review by MissingNO57 — DISMISSED** [review]
LGTM for now, but we will probably be refactoring the whole structure bit soon.

File: docs/source/ml/index.rst (hunk 1 above)
Comment (baykaner) [reply docs/source/ml/index.rst]: good point!

**Review by MissingNO57 — APPROVED** [review]
LGTM

**Pull request merged**